lib/polo/sql_translator.rb

```diff
@@ -49,45 +49,32 @@ def ignore_transform(inserts)
     end
 
     def raw_sql(record)
-      connection = ActiveRecord::Base.connection
-      attributes = record.attributes
-
-      keys = attributes.keys.map do |key|
-        "`#{key}`"
-      end
-
-      values = attributes.map do |key, value|
-        column = record.column_for_attribute(key)
-        connection.quote(cast_attribute(record, column, value))
-      end
-
-      "INSERT INTO `#{record.class.table_name}` (#{keys.join(', ')}) VALUES (#{values.join(', ')})"
+      record.class.arel_table.create_insert.tap do |insert_manager|
+        insert_manager.insert(insert_values(record))
+      end.to_sql
     end
 
-    module ActiveRecordLessThanFourPointTwo
-      def cast_attribute(record, column, value)
-        attribute = record.send(:type_cast_attribute_for_write, column, value)
-
-        if record.class.serialized_attributes.include?(column.name)
-          attribute.serialize
-        else
-          attribute
-        end
+    module ActiveRecordLessThanFour
+      def insert_values(record)
+        record.send(:arel_attributes_values)
       end
     end
 
-    module ActiveRecordFourPointTwoOrGreater
-      def cast_attribute(record, column, value)
-        column.type_cast_for_database(value)
+    module ActiveRecordFourOrGreater
+      def insert_values(record)
+        connection = ActiveRecord::Base.connection
+        values = record.send(:arel_attributes_with_values_for_create, record.attribute_names)
+        values.each do |attribute, value|
+          column = record.column_for_attribute(attribute.name)
+          values[attribute] = connection.type_cast(value, column)
+        end
```

Comment:
I think it looks alright. It's ok since this code is extracted into its own method.
It might be worth writing some docs for this `method`/`module` now that things are getting a little more complicated.